### PR TITLE
Pixel re-implemented to trigger immediately on visibility

### LINKED ISSUE #6984: Trigger amp-pixel directly, not via layout.

### PR DESCRIPTION
**Pull request by dvoytenko**
This will make `amp-pixel` to be more inline with an impression signal. The cost of this is very small since it only creates a single low-weight image.

Closes #6984
/cc @keithwrightbos @tdrl

### CODE REVIEW

File: builtins/amp-pixel.js [hunk 1]
```diff
@@ -16,59 +16,81 @@
...
+  trigger_() {
+    this.triggerPromise_ = Promise.resolve().then(() => {
       const src = this.element.getAttribute('src');
       return urlReplacementsForDoc(this.getAmpDoc())
-          .expandAsync(this.assertSource(src))
+          .expandAsync(this.assertSource_(src))
           .then(src => {
             const image = new Image();
             image.src = src;
             this.element.appendChild(image);
```
Comment (cramforce) [inline builtins/amp-pixel.js]: Maybe kill this line. It doesn't do anything useful.

File: builtins/amp-pixel.js [hunk 2]
```diff
@@ -16,59 +16,81 @@
...
 
-    /** @override */
-    buildCallback() {
-      // Consider the element invisible.
-      this.element.setAttribute('aria-hidden', 'true');
-    }
+  /** @override */
+  buildCallback() {
+    // Element is invisible.
+    toggle(this.element, false);
```
Comment (cramforce) [inline builtins/amp-pixel.js]: Why do this? Already `display:none`, right?

File: builtins/amp-pixel.js [hunk 3]
```diff
@@ -16,59 +16,81 @@
...
+  }
 
-    /** @override */
-    isLayoutSupported(layout) {
-      return layout == Layout.FIXED;
-    }
+  /** @override */
+  isLayoutSupported(layout) {
+    // "fixed" layout is supported, but in reality it's always width/height = 0.
+    return layout == Layout.FIXED;
```
Comment (jridgewell) [inline builtins/amp-pixel.js]: Is this necessary, now that we don't care about `#layoutCallback` (we're just waiting for `#buildCallback` now)?

File: builtins/amp-pixel.js (hunk 2 above)
Comment (dvoytenko) [inline builtins/amp-pixel.js]: Right. Fixed.

File: builtins/amp-pixel.js (hunk 3 above)
Comment (dvoytenko) [inline builtins/amp-pixel.js]: Yes. Still necessary. Static layout is assigned before `layoutCallback`. Actually, it's assigned before `buildCallback` too. And currently all `amp-pixel` are assumed to be `layout=fixed`.

File: builtins/amp-pixel.js (hunk 1 above)
Comment (dvoytenko) [inline builtins/amp-pixel.js]: Removed.

**Comment by dvoytenko** [comment]
@cramforce @jridgewell ptal

File: builtins/amp-pixel.js (hunk 3 above)
Comment (jridgewell) [inline builtins/amp-pixel.js]: I don't follow. How does a static layout change it?

Comment (dvoytenko) [inline builtins/amp-pixel.js]: This is the layout that's automatically calculated for all `amp-pixel` right now. Basically, all `amp-pixel` are "layout=fixed" unless explicitly overridden. And, of course, no one explicitly overrides that. Thus, here we have no choice but allow it. Alternatively, we'd get bunch of errors.

Comment (jridgewell) [inline builtins/amp-pixel.js]: But we could just allow any old value now? As in, the fixed positioning was added so they'd be laid out with a higher priority than normal content. Now that we don't care about layout, they could be anything.

Comment (dvoytenko) [inline builtins/amp-pixel.js]: It's not `position:fixed`, but `layout=fixed`. I could definitely allow all layouts, though - in reality, it doesn't matter here.

Comment (jridgewell) [inline builtins/amp-pixel.js]: Ohhhh!

**Pull request merged**